app/questionnaire/questionnaire_schema.py

```diff
@@ -1190,127 +1192,83 @@ def _populate_when_rules_section_dependencies(self) -> None:
         Question variants and content variants don't need including, since the answer ids, block ids, and question ids
         remain the same, so a change in the variant, does not impact questionnaire progress.
         """
-        progress_section_dependencies = (
-            self._when_rules_section_dependencies_by_section_for_progress_value_source
-        )
-
         for section in self.get_sections():
+            rules: list[Mapping] = []
             when_rules = get_values_for_key(
                 "when",
                 data=section,
                 ignore_keys=["question_variants", "content_variants"],
             )
-            rules: list = list(when_rules)
-
-            (
-                rules_section_dependencies,
-                rule_section_dependencies_for_progress_value_source,
-                rule_block_dependencies_for_progress_value_source,
-            ) = self._get_rules_section_dependencies(section["id"], rules)
+            for when_rule in when_rules:
+                rules.extend(get_mappings_with_key("source", data=when_rule))
 
-            if rules_section_dependencies:
-                self._when_rules_section_dependencies_by_section[
-                    section["id"]
-                ] = rules_section_dependencies
-
-            for (
-                key,
-                values,
-            ) in rule_section_dependencies_for_progress_value_source.items():
-                progress_section_dependencies[key].update(values)
-
-            self._populate_block_dependencies_for_progress_value_source(
-                rule_block_dependencies_for_progress_value_source
-            )
+            for rule in rules:
+                self._populate_dependencies_for_rule(
+                    rule, current_section_id=section["id"]
+                )
 
-    def _populate_block_dependencies_for_progress_value_source(
```

Comment:
V minor - the doc string in `_populate_when_rules_section_dependencies()` references this function and might need to be tweaked now?

Reply:
ah great shout - have updated the docstring